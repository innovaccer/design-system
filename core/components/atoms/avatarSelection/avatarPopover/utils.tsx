import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
