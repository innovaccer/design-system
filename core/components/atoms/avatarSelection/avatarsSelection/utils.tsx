import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
