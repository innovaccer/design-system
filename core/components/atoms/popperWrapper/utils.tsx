// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
