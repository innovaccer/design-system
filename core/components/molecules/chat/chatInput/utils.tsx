import { MessageType } from './ChatInput';

export const positionMentionPopup = (
  textareaRef: React.RefObject<HTMLDivElement>,
  setMentionPosition: React.Dispatch<React.SetStateAction<{ top: number; left: number }>>
) => {
  if (!textareaRef.current) return;

  const selection = window.getSelection();
  if (!selection || !selection.rangeCount) return;

  const range = selection.getRangeAt(0);
  const rect = range.getBoundingClientRect();
  const editorRect = textareaRef.current.getBoundingClientRect();

  // Calculate position based on the cursor position
  let top = rect.bottom + window.scrollY;
  let left = rect.left + window.scrollX;

  // Adjust position if it would go off-screen
  if (top + 150 > window.innerHeight) {
    top = rect.top + window.scrollY - 150;
  }
  if (left + 200 > window.innerWidth) {
    left = window.innerWidth - 200;
  }

  // Adjust position relative to the editor
  top -= editorRect.top + window.scrollY - 12;
  left -= editorRect.left + window.scrollX - 8;

  setMentionPosition({ top, left });
};

export const extractMessageData = (textareaRef: React.RefObject<HTMLDivElement>): MessageType[] => {
  const elements = textareaRef.current?.childNodes;

  return Array.from(elements || []).map((item: any) => {
    if (item?.dataset?.type === 'mention') {
      const mentionValue = JSON.parse(item.dataset.content);
      return { type: 'mention', content: mentionValue?.data, nodeName: item.nodeName };
    }

    // handles new line by pressing enter
    else if (item.innerText === '\n' && item.nodeName === 'DIV') {
      return { type: 'newline', content: '\n', nodeName: item.nodeName };
    }

    // handles line break by pressing shift + enter
    else if (item.nodeName === 'BR') {
      return { type: 'linebreak', content: '<br>', nodeName: item.nodeName };
    }

    return { type: 'text', content: item.innerText || item.textContent, nodeName: item.nodeName };
  });
};
