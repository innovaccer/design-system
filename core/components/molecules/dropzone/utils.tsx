const isIe = (userAgent: string) => {
  return userAgent.indexOf('MSIE') !== -1 || userAgent.indexOf('Trident/') !== -1;
};

const isEdge = (userAgent: string) => {
  return userAgent.indexOf('Edge/') !== -1;
};

export const accepts = (file: File, acceptedFiles?: string | string[]) => {
  if (file && acceptedFiles) {
    const acceptedFilesArray = Array.isArray(acceptedFiles) ? acceptedFiles : acceptedFiles.split(',');
    const fileName = file.name || '';
    const mimeType = (file.type || '').toLowerCase();
    const baseMimeType = mimeType.replace(/\/.*$/, '');

    if (!mimeType) {
      return true;
    }

    return acceptedFilesArray.some((type) => {
      const validType = type.trim().toLowerCase();
      if (validType.charAt(0) === '.') {
        return fileName.toLowerCase().endsWith(validType);
      }
      if (validType.endsWith('/*')) {
        return baseMimeType === validType.replace(/\/.*$/, '');
      }
      return mimeType === validType;
    });
  }
  return true;
};

export const isPropagationStopped = (event: any) => {
  if (typeof event.isPropagationStopped === 'function') {
    return event.isPropagationStopped();
  }
  if (typeof event.cancelBubble !== 'undefined') {
    return event.cancelBubble;
  }
  return false;
};

export const isEvtWithFiles = (event: any) => {
  if (!event.dataTransfer) {
    return !!event.target && !!event.target.files;
  }

  return Array.prototype.some.call(
    event.dataTransfer.types,
    (type) => type === 'Files' || type === 'application/x-moz-file'
  );
};

export const isKindFile = (item: null | Record<string, unknown>) => {
  return typeof item === 'object' && item !== null && item.kind === 'file';
};

export const onDocumentDragOver = (event: DragEvent) => {
  event.preventDefault();
};

export const isIeOrEdge = (userAgent = window.navigator.userAgent) => {
  return isIe(userAgent) || isEdge(userAgent);
};

type functype = (event: Event, ...args: unknown[]) => void;

export const composeEventHandlers = (...fns: functype[]) => {
  return (event: Event, ...args: unknown[]) =>
    fns.some((fn) => {
      if (!isPropagationStopped(event) && fn) {
        fn(event, ...args);
      }
      return isPropagationStopped(event);
    });
};

export const reducer = (state: any, action: any) => {
  switch (action.type) {
    case 'focus':
      return {
        ...state,
        isFocused: true,
      };
    case 'blur':
      return {
        ...state,
        isFocused: false,
      };
    case 'openDialog':
      return {
        ...state,
        isFileDialogActive: true,
      };
    case 'closeDialog':
      return {
        ...state,
        isFileDialogActive: false,
      };
    case 'setDraggedFiles':
      /* eslint no-case-declarations: 0 */
      const { isDragActive, draggedFiles } = action;
      return {
        ...state,
        draggedFiles,
        isDragActive,
      };
    case 'setFiles':
      return {
        ...state,
        acceptedFiles: action.acceptedFiles,
        fileRejections: action.fileRejections,
      };
    case 'reset':
      return {
        ...state,
        isFileDialogActive: false,
        isDragActive: false,
        draggedFiles: [],
        acceptedFiles: [],
        fileRejections: [],
      };
    default:
      return state;
  }
};
