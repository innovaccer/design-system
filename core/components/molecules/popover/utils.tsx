import { Placement } from '@floating-ui/react';
import { PositionType as Position } from '@/common.type';

export const mapPositionToPlacement = (position: Position): Placement => {
  const positionMap: Record<Position, Placement> = {
    auto: 'bottom',
    'auto-start': 'bottom-start',
    'auto-end': 'bottom-end',
    'top-start': 'top-start',
    top: 'top',
    'top-end': 'top-end',
    'right-start': 'right-start',
    right: 'right',
    'right-end': 'right-end',
    'bottom-end': 'bottom-end',
    bottom: 'bottom',
    'bottom-start': 'bottom-start',
    'left-end': 'left-end',
    left: 'left',
    'left-start': 'left-start',
  };

  return positionMap[position] || 'bottom';
};
