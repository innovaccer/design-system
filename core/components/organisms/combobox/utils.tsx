import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  inputTriggerRef?: any,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  multiSelect?: boolean,
  listRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef);
      break;
    case 'Enter':
      handleEnterKey(focusedOption, multiSelect, inputTriggerRef, listRef, setFocusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      inputTriggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (
  focusedOption: Element | undefined,
  multiSelect?: boolean,
  inputTriggerRef?: any,
  listRef?: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>
) => {
  (focusedOption as HTMLElement)?.click();
  if (!multiSelect) {
    inputTriggerRef.current.focus();
  } else {
    // to focus first option by default when last option is selected
    const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
    const listArr = Array.from(listItems);
    const index = listArr.findIndex((item) => {
      return item == focusedOption;
    });

    if (index === listArr.length - 1) {
      (listItems[0] as HTMLElement).focus();
      setFocusedOption && setFocusedOption(listItems[0]);
      listItems[0].scrollIntoView({ block: 'center' });
    }
  }
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];

  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption?.scrollIntoView?.({ block: 'center' });
};
