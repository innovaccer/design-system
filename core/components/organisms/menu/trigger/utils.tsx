import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    case 'ArrowDown':
      event.preventDefault();
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'Escape':
    case 'Tab':
      setOpenPopover?.(false);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();

  if (targetOption && typeof targetOption.scrollIntoView === 'function') {
    (targetOption as HTMLElement)?.scrollIntoView({ block: 'end' });
  }
  setFocusedOption && setFocusedOption(targetOption);
};
