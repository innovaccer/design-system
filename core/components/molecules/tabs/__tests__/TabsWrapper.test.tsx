import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import { Tab, TabsWrapper, Text } from '@/index';
import { TabsWrapperProps as Props } from '@/index.type';

const TabValue = [0, 1];
const FunctionValue = jest.fn();
const Label = <Text>Tab(Recommended)</Text>;

describe('TabsWrapper component', () => {
  const Mapper: Record<string, any> = {
    activeTab: valueHelper(TabValue, { required: true, iterate: true }),
    onTabChange: valueHelper(FunctionValue, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { asFragment } = render(
        <TabsWrapper
          {...attr}
        >
          <Tab label={Label}>Tab 1</Tab>
          <Tab label={Label} disabled={true}>Tab 2</Tab>
        </TabsWrapper>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  };

  testHelper(Mapper, testFunc);
});

describe('TabsWrapper component', () => {
  it('renders single tab', () => {
    const activeTab = 0;
    const activeTabContent = 'Tab 1';

    const { getByTestId, getAllByTestId } = render(
      <TabsWrapper active={activeTab} onTabChange={FunctionValue}>
        <Tab label={<></>}>Tab 1</Tab>
      </TabsWrapper>
    );

    expect(getByTestId('DesignSystem-Tabs--Content').textContent).toMatch(activeTabContent);
    expect(getAllByTestId('DesignSystem-Tabs--Header')[activeTab]).toHaveClass('Tab--active');
  });

  it('renders active tab and children', () => {
    const activeTab = 1;
    const activeTabContent = 'Tab 2';

    const { getByTestId, getAllByTestId } = render(
      <TabsWrapper active={activeTab} onTabChange={FunctionValue}>
        <Tab label={<></>}>Tab 1</Tab>
        <Tab label={<></>}>Tab 2</Tab>
      </TabsWrapper>
    );

    expect(getByTestId('DesignSystem-Tabs--Content').textContent).toMatch(activeTabContent);
    expect(getAllByTestId('DesignSystem-Tabs--Header')[activeTab]).toHaveClass('Tab--active');
  });

  it('renders default tab as 0', () => {
    const defaultTab = 0;
    const defaultTabContent = 'Tab 1';

    const { getByTestId, getAllByTestId } = render(
      <TabsWrapper onTabChange={FunctionValue}>
        <Tab label={<></>}>Tab 1</Tab>
        <Tab label={<></>}>Tab 2</Tab>
      </TabsWrapper>
    );

    expect(getByTestId('DesignSystem-Tabs--Content').textContent).toMatch(defaultTabContent);
    expect(getAllByTestId('DesignSystem-Tabs--Header')[defaultTab]).toHaveClass('Tab--active');
  });

  it('default tab is 0 if activeTab > total tabs length', () => {
    const activeTab = 3;
    const activeTabContent = 'Tab 1';
    const defaultActiveTab = 0;

    const { getByTestId, getAllByTestId } = render(
      <TabsWrapper active={activeTab} onTabChange={FunctionValue}>
        <Tab label={<></>}>Tab 1</Tab>
        <Tab label={<></>}>Tab 2</Tab>
      </TabsWrapper>
    );

    expect(getByTestId('DesignSystem-Tabs--Content').textContent).toMatch(activeTabContent);
    expect(getAllByTestId('DesignSystem-Tabs--Header')[defaultActiveTab]).toHaveClass('Tab--active');
  });

  it('renders disabled tab', () => {
    const activeTab = 1;
    const disabledTab = 1;

    const { getAllByTestId } = render(
      <TabsWrapper active={activeTab} onTabChange={FunctionValue}>
        <Tab label={<></>}>Tab 1</Tab>
        <Tab label={<></>} disabled={true}>Tab 2</Tab>
      </TabsWrapper>
    );

    expect(getAllByTestId('DesignSystem-Tabs--Header')[activeTab]).not.toHaveClass('Tab--active');
    expect(getAllByTestId('DesignSystem-Tabs--Header')[disabledTab]).toHaveClass('Tab--disabled');
  });

});

describe('TabsWrapper component with prop: onTabChange', () => {
  const tabs = (tab: number) => (
    <TabsWrapper active={tab} onTabChange={FunctionValue}>
      <Tab label={<div data-test="DesignSystem-TabLabel-One">Label 1</div>}>Tab 1</Tab>
      <Tab label={<div data-test="DesignSystem-TabLabel-Two">Label 2</div>}>Tab 2</Tab>
    </TabsWrapper>
  );

  afterEach(() => jest.clearAllMocks());

  it('TabsWrapper component with onTabChange callback on click', () => {
    const activeTab = 0;
    const clickedTab = 1;

    const { getByTestId, getAllByTestId, rerender } = render(tabs(activeTab));

    const tab = getByTestId('DesignSystem-TabLabel-Two');
    fireEvent.click(tab);

    expect(FunctionValue).toHaveBeenCalledWith(clickedTab);
    rerender(tabs(clickedTab));
    expect(getAllByTestId('DesignSystem-Tabs--Header')[clickedTab]).toHaveClass('Tab--active');
  });

  it('TabsWrapper component with onTabChange callback on enter press', () => {
    const activeTab = 0;
    const KeyedTab = 1;

    const { getByTestId, getAllByTestId, rerender } = render(tabs(activeTab));

    const tab = getByTestId('DesignSystem-TabLabel-Two');
    fireEvent.keyDown(tab, { key: 'Enter', keyCode: 13 });

    expect(FunctionValue).toHaveBeenCalledWith(KeyedTab);
    rerender(tabs(KeyedTab));
    expect(getAllByTestId('DesignSystem-Tabs--Header')[KeyedTab]).toHaveClass('Tab--active');
  });

  it('TabsWrapper component with onTabChange callback not called on any other key', () => {
    const activeTab = 0;
    const KeyedTab = 1;

    const { getByTestId, getAllByTestId, rerender } = render(tabs(activeTab));
    const tab = getByTestId('DesignSystem-TabLabel-Two');
    fireEvent.keyDown(tab, { key: 'Space', keyCode: 32 });

    expect(FunctionValue).not.toHaveBeenCalledWith(KeyedTab);
    rerender(tabs(activeTab));
    expect(getAllByTestId('DesignSystem-Tabs--Header')[KeyedTab]).not.toHaveClass('Tab--active');
  });

  it('TabsWrapper component changes tab right when ArrowRight is pressed', () => {
    const activeTab = 0;

    const { getByTestId } = render(tabs(activeTab));
    const firstLabel = getByTestId('DesignSystem-TabLabel-One');
    const secondLabel = getByTestId('DesignSystem-TabLabel-Two');

    fireEvent.focus(firstLabel);
    fireEvent.keyDown(firstLabel, { key: 'ArrowRight', keyCode: 39 });

    expect(document.activeElement).toContainElement(secondLabel);
  });

  it('TabsWrapper component changes tab left when ArrowLeft is pressed', () => {
    const activeTab = 1;

    const { getByTestId } = render(tabs(activeTab));
    const firstLabel = getByTestId('DesignSystem-TabLabel-One');
    const secondLabel = getByTestId('DesignSystem-TabLabel-Two');

    fireEvent.focus(secondLabel);
    fireEvent.keyDown(secondLabel, { key: 'ArrowLeft', keyCode: 37 });

    expect(document.activeElement).toContainElement(firstLabel);
  });
});
