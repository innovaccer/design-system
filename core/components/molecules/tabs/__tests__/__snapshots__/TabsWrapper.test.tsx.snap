// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsWrapper component 
  activeTab: 0, onTabChange: "[Function]"
 1`] = `
<DocumentFragment>
  <div
    class="TabsWrapper"
    data-test="DesignSystem-TabsWrapper"
  >
    <div
      class="TabsWrapper-header"
    >
      <div
        class="Tab Tab--active"
        data-test="DesignSystem-Tabs--Header"
        tabindex="0"
      >
        <span
          class="Text Text--default Text--regular"
          data-test="DesignSystem-Text"
        >
          Tab(Recommended)
        </span>
      </div>
      <div
        class="Tab Tab--disabled"
        data-test="DesignSystem-Tabs--Header"
        tabindex="-1"
      >
        <span
          class="Text Text--default Text--regular"
          data-test="DesignSystem-Text"
        >
          Tab(Recommended)
        </span>
      </div>
    </div>
    <div
      class="TabsWrapper-content"
      data-test="DesignSystem-Tabs--Content"
    >
      Tab 1
    </div>
  </div>
</DocumentFragment>
`;

exports[`TabsWrapper component 
  activeTab: 1, onTabChange: "[Function]"
 1`] = `
<DocumentFragment>
  <div
    class="TabsWrapper"
    data-test="DesignSystem-TabsWrapper"
  >
    <div
      class="TabsWrapper-header"
    >
      <div
        class="Tab Tab--active"
        data-test="DesignSystem-Tabs--Header"
        tabindex="0"
      >
        <span
          class="Text Text--default Text--regular"
          data-test="DesignSystem-Text"
        >
          Tab(Recommended)
        </span>
      </div>
      <div
        class="Tab Tab--disabled"
        data-test="DesignSystem-Tabs--Header"
        tabindex="-1"
      >
        <span
          class="Text Text--default Text--regular"
          data-test="DesignSystem-Text"
        >
          Tab(Recommended)
        </span>
      </div>
    </div>
    <div
      class="TabsWrapper-content"
      data-test="DesignSystem-Tabs--Content"
    >
      Tab 1
    </div>
  </div>
</DocumentFragment>
`;
